staging/src/k8s.io/apiserver/pkg/registry/rest/validate.go

```diff
@@ -106,3 +111,171 @@ func parseSubresourcePath(subresourcePath string) ([]string, error) {
+// incrementing the panic metric, and logging an error message
+// if DeclarativeValidationTakeover=disabled, and adding a validation error if enabled.
+func WithRecoverUpdate(
+	validateUpdateFunc func(ctx context.Context, options sets.Set[string], scheme *runtime.Scheme, obj, oldObj runtime.Object) field.ErrorList,
+) func(ctx context.Context, options sets.Set[string], scheme *runtime.Scheme, obj, oldObj runtime.Object) field.ErrorList {
+	return func(ctx context.Context, options sets.Set[string], scheme *runtime.Scheme, obj, oldObj runtime.Object) (errs field.ErrorList) {
+		defer createDeclarativeValidationPanicHandler(&errs)()
+
+		// Call update validation
+		errs = validateUpdateFunc(ctx, options, scheme, obj, oldObj)
+
+		return errs
+	}
+}
+
+// ValidateDeclarativelyWithRecover validates obj against declarative validation tags
+// with panic recovery logic. It uses the API version extracted from ctx and the
+// provided scheme for validation.
+//
+// The ctx MUST contain requestInfo, which determines the target API for
+// validation. The obj is converted to the API version using the provided scheme
+// before validation occurs. The scheme MUST have the declarative validation
+// registered for the requested resource/subresource.
+//
+// option should contain any validation options that the declarative validation
+// tags expect.
+//
+// Returns a field.ErrorList containing any validation errors. An internal error
+// is included if requestInfo is missing from the context, if version
+// conversion fails, or if a panic occurs during validation when
+// DeclarativeValidationTakeover is enabled.
+func ValidateDeclarativelyWithRecovery(ctx context.Context, options sets.Set[string], scheme *runtime.Scheme, obj runtime.Object) field.ErrorList {
+	return WithRecover(ValidateDeclaratively)(ctx, nil, scheme, obj)
+}
+
+// ValidateUpdateDeclarativelyWithRecover validates obj and oldObj against declarative
+// validation tags with panic recovery logic. It uses the API version extracted from
+// ctx and the provided scheme for validation.
+//
+// The ctx MUST contain requestInfo, which determines the target API for
+// validation. The obj is converted to the API version using the provided scheme
+// before validation occurs. The scheme MUST have the declarative validation
+// registered for the requested resource/subresource.
+//
+// option should contain any validation options that the declarative validation
+// tags expect.
+//
+// Returns a field.ErrorList containing any validation errors. An internal error
+// is included if requestInfo is missing from the context, if version
+// conversion fails, or if a panic occurs during validation when
+// DeclarativeValidationTakeover is enabled.
+func ValidateUpdateDeclarativelyWithRecovery(ctx context.Context, options sets.Set[string], scheme *runtime.Scheme, obj, oldObj runtime.Object) field.ErrorList {
+	return WithRecoverUpdate(ValidateUpdateDeclaratively)(ctx, nil, scheme, obj, oldObj)
```

Comment:
```suggestion
	return WithRecoverUpdate(ValidateUpdateDeclaratively)(ctx, options, scheme, obj, oldObj)
```

Reply:
Apologies, thanks for catching this, fixed now